dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/clouddumper/DriverRetriever.java

```diff
@@ -144,15 +141,18 @@ public Optional<Path> getDriver(String name) throws IOException {
       LOG.info("Got no driver for connector '{}'.", name);
       return Optional.empty();
     }
-    DriverInformation driverInformation = driverInformationMap.get(name);
+    return this.downloader.apply(driverInformationMap.get(name));
+  }
+
+  private Optional<Path> download(DriverInformation driverInformation) throws IOException {
```

Comment:
When does `download` return empty Optional?

Reply:
When there is no driver needed for the connector.

Reply:
The `getDriver` method can return empty Optional, but I don't see how the `download` method could.

If I'm not mistaken, by looking at the code I think that the `download` method executes HTTP request always, so the method should always return non-empty Optional (or throw an exception). If this is the behavior, then can the return type be changed from `Optional<Path>` to `Path`?

Reply:
You are right. I initially wanted to have the memoization at an earlier point. Here, it does not make sense to have the memoization wrapped in `Optional` as it will always be populated. I've adapted it accordingly.